apps/web/pages/api/book/recurring-event.ts

```diff
@@ -53,16 +56,34 @@ async function handler(req: NextApiRequest & { userId?: number }, res: NextApiRe
       ...booking,
       appsStatus,
       allRecurringDates,
+      thirdPartyRecurringEventId,
+      numSlotsToCheckForAvailability,
       currentRecurringIndex: key,
       noEmail: key !== 0,
     };
 
-    const eachRecurringBooking = await handleNewBooking(recurringEventReq, {
-      isNotAnApiCall: true,
-      skipAvailabilityCheck: key >= numSlotsToCheckForAvailability,
-    });
+    const promiseEachRecurringBooking: ReturnType<typeof handleNewBooking> = handleNewBooking(
+      recurringEventReq,
+      {
+        isNotAnApiCall: true,
+        skipAvailabilityCheck: key >= numSlotsToCheckForAvailability || thirdPartyRecurringEventId !== null,
```

Comment:
Also, I wanted to point out that for the calendar apps that don't implement recurring api(all except Google Calendar) at the moment, `thirdPartyRecurringEventId` comes as undefined which is not really a value but `skipAvailabilityCheck` is correctly calculated because `null !== undefined`. 

We should explicitly handle this.